Riot/Modules/Room/RoomViewController.m

```diff
@@ -4979,10 +4979,31 @@ - (void)roomTitleView:(RoomTitleView*)titleView recognizeTapGesture:(UITapGestur
     }
     else if (tappedView == previewHeader.leftButton)
     {
-        [self declineRoomInvitation];
+        [self presentDeclineOptions];
     }
 }
 
+- (void)presentDeclineOptions
+{
+    UIAlertController *actionSheet = [UIAlertController alertControllerWithTitle:[VectorL10n roomPreviewDeclineInvitationOptions]
+                                                                         message:nil
+                                                                  preferredStyle:UIAlertControllerStyleActionSheet];
+    [actionSheet addAction:[UIAlertAction actionWithTitle:[VectorL10n decline]
+                                                    style:UIAlertActionStyleDefault
+                                                  handler:^(UIAlertAction * _Nonnull action) {
+        [self declineRoomInvitation];
+    }]];
+    [actionSheet addAction:[UIAlertAction actionWithTitle:[VectorL10n ignoreUser]
+                                                    style:UIAlertActionStyleDestructive
+                                                  handler:^(UIAlertAction * _Nonnull action) {
+        [self ignoreInviteSender];
+    }]];
+    [actionSheet addAction:[UIAlertAction actionWithTitle:[VectorL10n cancel]
+                                                    style:UIAlertActionStyleCancel
+                                                  handler:nil]];
+    [self presentViewController:actionSheet animated:YES completion:nil];
```

Comment:
As this is a sheet, I think you might need to handle popover presentation on iPad?
```
actionSheet.popoverPresentationController.sourceView = …
actionSheet.popoverPresentationController.sourceRect = …
```

Reply:
Thanks, so easy to forget about this death-trap of an API

![Simulator Screen Shot - iPad Pro (12 9-inch) (5th generation) - 2022-03-14 at 08 44 04](https://user-images.githubusercontent.com/3922159/158136864-60c4d046-a6ae-4491-85df-541da693a04b.png)

Reply:
Yeah, it's pretty annoying this API can't be checked at compile time.